shell/utils/validators/formRules/index.ts

```diff
@@ -458,6 +458,8 @@ export default function(t: Translation, { key = 'Value' }: ValidationOptions): {
     return runValidators(val, [startHyphen('label'), endHyphen('label'), startDot('label'), endDot('label'), required]);
   };
 
+  const repo: Validator = (val) => val && !/((git|ssh|http(s)?)|(git@[\w\.]+))(\:(\/\/)?)([\w\.@\:/\-~]+)(\.git)?(\/)?/.test(val) ? t('validation.setting.repo', { key }) : undefined;
```

Comment:
I think this validation part is not a vue3 regression, and it might be good to have a unit test to try different cases against this regex